File: android-lib/src/androidTest/java/com/kinvey/androidTest/store/data/network/AndroidNetworkManagerTest.java
package com.kinvey.androidTest.store.data.network;

import android.content.Context;

import androidx.test.ext.junit.runners.AndroidJUnit4;
import androidx.test.filters.SmallTest;
import androidx.test.platform.app.InstrumentationRegistry;

import com.kinvey.android.Client;
import com.kinvey.android.network.AndroidNetworkManager;
import com.kinvey.androidTest.model.Person;

import org.junit.After;
import org.junit.Test;
import org.junit.runner.RunWith;

import static junit.framework.Assert.assertNotNull;
import static junit.framework.Assert.assertTrue;

@RunWith(AndroidJUnit4.class)
@SmallTest
public class AndroidNetworkManagerTest {

    private Client client;

    @After
    public void tearDown() {
        client.performLockDown();
        if (client.getKinveyHandlerThread() != null) {
            try {
                client.stopKinveyHandlerThread();
            } catch (Throwable throwable) {
                throwable.printStackTrace();
            }
        }
    }

    @Test
    public void testIsOnline() {
        Context mMockContext = InstrumentationRegistry.getInstrumentation().getTargetContext();
        client = new Client.Builder(mMockContext).build();
        AndroidNetworkManager androidNetworkManager = new AndroidNetworkManager<>(Person.COLLECTION, Person.class, client);
        assertNotNull(androidNetworkManager);
        assertTrue(androidNetworkManager.isOnline());
    }

}

File: android-lib/src/androidTest/java/com/kinvey/androidTest/store/data/network/AndroidNetworkManagerTest.kt
package com.kinvey.androidTest.store.data.network

import android.content.Context
import androidx.test.filters.SmallTest
import androidx.test.platform.app.InstrumentationRegistry
import androidx.test.runner.AndroidJUnit4
import com.kinvey.android.Client
import com.kinvey.android.Client.Builder
import com.kinvey.android.model.User
import com.kinvey.android.network.AndroidNetworkManager
import com.kinvey.androidTest.model.Person
import junit.framework.Assert.assertNotNull
import junit.framework.Assert.assertTrue
import org.junit.After
import org.junit.Test
import org.junit.runner.RunWith

@RunWith(AndroidJUnit4::class)
@SmallTest
class AndroidNetworkManagerTest {

    private var client: Client<*>? = null

    @After
    fun tearDown() {
        client?.performLockDown()
        if (Client.kinveyHandlerThread != null) {
            try {
                client?.stopKinveyHandlerThread()
            } catch (throwable: Throwable) {
                throwable.printStackTrace()
            }
        }
    }

    @Test
    fun testIsOnline() {
        val mMockContext: Context? = InstrumentationRegistry.getInstrumentation().targetContext
        client = Builder<User>(mMockContext).build()
        val androidNetworkManager = AndroidNetworkManager(Person.COLLECTION, Person::class.java, client)
        assertNotNull(androidNetworkManager)
        assertTrue(androidNetworkManager.isOnline)
    }
}